src/sys/unix.rs

```diff
@@ -1773,6 +1803,41 @@ impl crate::Socket {
                 })
         }
     }
+
+    /// Attach Berkeley Packet Filter(BPF) on this socket.
+    ///
+    /// BPF allows a user-space program to attach a filter onto any socket
+    /// and allow or disallow certain types of data to come through the socket.
+    ///
+    /// For more information about this option, see [filter](https://www.kernel.org/doc/html/v5.12/networking/filter.html)
+    #[cfg(all(feature = "all", target_os = "linux"))]
+    pub fn attach_filter(&self, mut filters: Vec<SockFilter>) -> io::Result<()> {
+        filters.shrink_to_fit();
+        assert_eq!(filters.capacity(), filters.len());
+
+        let prog = SockFprog {
+            len: filters.len() as u16,
+            filter: filters.as_mut_ptr(),
+        };
+        forget(filters);
+
+        unsafe {
+            setsockopt(
+                self.as_raw(),
+                libc::SOL_SOCKET,
+                libc::SO_ATTACH_FILTER,
+                prog,
```

Comment:
Pretty sure this needs to be `&prog`.

Reply:
The payload has been converted to a pointer at: 

https://github.com/rust-lang/socket2/blob/5c7b03f61ef9ddb21233a18f643d6b9367429c74/src/sys/unix.rs#L963

I tried to change to `&prog` and it did not run successfully as before.

Reply:
You're right, I should really try to remember the code I wrote :)